vendor/bat-native-ledger/src/bat/ledger/internal/publisher/publisher_status_helper.cc

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/publisher/publisher_status_helper.h"
+
+#include <algorithm>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/time/time.h"
+#include "bat/ledger/internal/ledger_impl.h"
+
+namespace {
+
+struct PublisherStatusData {
+  ledger::PublisherStatus status;
+  uint64_t updated_at;
+};
+
+using PublisherStatusMap = std::map<std::string, PublisherStatusData>;
+
+struct RefreshTaskInfo {
+  RefreshTaskInfo(
+      bat_ledger::LedgerImpl* ledger,
+      PublisherStatusMap&& status_map,
+      std::function<void(PublisherStatusMap)> callback)
+      : ledger(ledger),
+        map(std::move(status_map)),
+        current(map.begin()),
+        callback(callback) {}
+
+  bat_ledger::LedgerImpl* ledger;
+  PublisherStatusMap map;
+  PublisherStatusMap::iterator current;
+  std::function<void(PublisherStatusMap)> callback;
+};
+
+void RefreshNext(std::shared_ptr<RefreshTaskInfo> task_info) {
+  // Find the first map element that has an expired status.
+  task_info->current = std::find_if(
```

Comment:
check if `task_info` is null

Reply:
I've added a DCHECK since it should never be empty.